src/test/java/org/assertj/core/internal/throwables/Throwables_assertHasRootCause_Test.java

```diff
@@ -0,0 +1,197 @@
+    Throwable expected = new IllegalArgumentException("bang");
+    Throwable throwable = withRootCause(rootCause);
+
+    // WHEN
+    throwables.assertHasRootCause(someInfo(), throwable, expected);
+
+    // THEN
+    // no exception thrown
+  }
+
+  @Test
+  public void should_pass_if_root_cause_is_expected_cause() {
+    // GIVEN
+    Throwable cause = new IllegalArgumentException("wibble");
+    Throwable expected = new IllegalArgumentException("wibble");
+    Throwable throwable = withCause(cause);
+
+    // WHEN
+    throwables.assertHasRootCause(someInfo(), throwable, expected);
+
+    // THEN
+    // no exception thrown
+  }
+
+  @Test
+  public void should_pass_if_actual_has_no_cause_and_expected_cause_is_null() {
+    // GIVEN
+    Throwable throwable = new Throwable();
+    Throwable expected = null;
+
+    // WHEN
+    throwables.assertHasRootCause(someInfo(), throwable, expected);
+
+    // THEN
+    // no exception thrown
+  }
+
+  @Test
+  public void should_fail_if_actual_is_null() {
+    // GIVEN
+    final Throwable throwable = null;
+    final Throwable expected = new Throwable();
+
+    // WHEN
+    AssertionError actual = expectAssertionError(
+      () -> throwables.assertHasRootCause(someInfo(), throwable, expected)
+    );
+
+    // THEN
+    assertThat(actual)
+      .hasMessage(actualIsNull());
+  }
+
+  @Test
+  public void should_fail_if_actual_has_no_cause_and_expected_cause_is_not_null() {
+    // GIVEN
+    final Throwable throwable = new Throwable();
+    final Throwable expected = new RuntimeException();
+    final AssertionInfo info = someInfo();
+
+    // WHEN
+    expectAssertionError(
+      () -> throwables.assertHasRootCause(info, throwable, expected)
+    );
+
+    // THEN
+    verify(failures).failure(info, shouldHaveRootCause(null, expected));
+  }
+
+  @Test
+  public void should_fail_if_actual_has_cause_and_expected_cause_is_null() {
+    // GIVEN
+    Throwable rootCause = new IllegalArgumentException();
+    final Throwable throwable = withRootCause(rootCause);
+    final Throwable expected = null;
+    final AssertionInfo info = someInfo();
+
+    // WHEN
+    expectAssertionError(
+      () -> throwables.assertHasRootCause(info, throwable, expected)
+    );
+
+    // THEN
+    verify(failures).failure(info, shouldHaveNoCause(throwable));
+  }
+
+  @Test
+  public void should_fail_if_cause_is_not_instance_of_expected_type() {
+    // GIVEN
+    Throwable rootCause = new IllegalArgumentException();
+    final Throwable throwable = withRootCause(rootCause);
+    final Throwable expected = new NullPointerException();
+    final AssertionInfo info = someInfo();
+
+    // WHEN
+    expectAssertionError(
+      () -> throwables.assertHasRootCause(info, throwable, expected)
+    );
+
+    // THEN
+    verify(failures).failure(info, shouldHaveRootCause(rootCause, expected));
+  }
+
+  @Test
+  public void should_fail_if_cause_has_not_the_expected_message() {
```

Comment:
`should_fail_if_cause` should be `should_fail_if_root_cause` to make it clear we are checking the root cause.
there are other tests that can be renamed that way too ;-)